v-next/hardhat-keystore/src/methods.ts

```diff
@@ -0,0 +1,93 @@
+import chalk from "chalk";
+
+import { io } from "./io.js";
+import { isAuthorized } from "./password-manager.js";
+import {
+  addNewSecret,
+  getKeystore,
+  removeKey,
+  setupKeystore,
+  validateKey,
+} from "./utils.js";
+
+export const PLUGIN_ID = "hardhat-keystore";
+
+export async function set(key: string, force: boolean = false): Promise<void> {
+  const keystore = await getKeystore();
+
+  if (keystore === undefined) {
+    await setupKeystore();
+  }
+
+  if (!validateKey(key)) {
+    return;
+  }
+
+  if ((await isAuthorized()) === false) {
```

Comment:
What is `isAuthorized` adding here? My expectation is there is no need for authorization, the user either provides a valid password for updating the keystore or doesn't?

Reply:
It's a placeholder. It will be implemented/modified as soon as we have the specs for the encryption